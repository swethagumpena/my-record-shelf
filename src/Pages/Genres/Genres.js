/* eslint-disable max-len */
/* eslint-disable react/prop-types */
import React from 'react';
import { useHistory } from 'react-router-dom';
import Card from '../../components/Card/Card';
import styles from './Genres.module.scss';

// eslint-disable-next-line no-unused-vars
const Genres = ({ filteredSongs }) => {
  const history = useHistory();
  const songsByCategory = Object.keys(filteredSongs).map((category) => (
    <div key={category}>
      <div className={styles.genreHeader}>
        <div className={styles.genreImage}>
          <img src={`assets/genre-${category}.png`} alt="genre-img" />
        </div>
        <div className={styles.songsContainer}>
          {category}
        </div>
      </div>

      <div className={styles.card}>
        {filteredSongs[category].map((eachSong) => (
          <Card
            key={eachSong.id}
            id={eachSong.id}
            albumArtUrl={eachSong.albumArtUrl}
            songName={eachSong.name}
            genreName={eachSong.genre.name}
            artistName={eachSong.artist.name}
          />
        ))}
      </div>
    </div>
  ));

  return (
    <div>
      <div className={styles.header}>
        <p className={styles.heading}>genres</p>
        <button className={styles.toggle} type="button" onClick={() => history.push('/songs')}>X</button>
      </div>
      <div data-testid="home" className="songs">
        {songsByCategory}
      </div>
    </div>

  );
};

// {
//   return (
//     <div>
//       <div className={styles.header}>
//         <p className={styles.heading}>genres</p>
//         <button className={styles.toggle} type="button" onClick={() => history.push('/songs')}>X</button>
//       </div>

//       {/* <div className={styles.songsContainer}>
//           <div className={styles.card}>
//             {songs.map((song) => (
//               <Card
//                 key={song.id}
//                 id={song.id}
//                 albumArtUrl={song.albumArtUrl}
//                 songName={song.name}
//                 genreName={song.genre.name}
//                 artistName={song.artist.name}
//               />
//             ))}
//           </div>
//         </div> */}

//     </div>
//   );
// };

export default Genres;
