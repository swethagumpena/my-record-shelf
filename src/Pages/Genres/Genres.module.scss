@import "../../styles/styles.scss";


.header{
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.heading{
    padding: 2rem 2rem 2rem 4rem;
    font-family: 'Poppins', sans-serif;
    font-weight: bolder;
    font-size: 2rem;
}

p{
    margin: 0;
}

.toggle{
    margin-right: 5rem;
}

.songsContainer {
    display: flex;
    background-color:#4C63AA;
    padding: 0.75rem 2rem;
    border-radius: 3rem;
    color: white;
    font-family: 'Poppins', sans-serif;
    font-weight: bolder;
    margin-left: -3rem;
}

.card {
    display: flex;
    flex-wrap: wrap;
    padding: 1rem;
    max-width: 100rem;
    padding: 0;
    justify-content: center;
}

.genreHeader{
    display: flex;
    align-items: center;
}

.genreImage{
    margin: 1rem 0 1rem 3rem;
}

@include media-mixin($mobile-breakpoint){
    .genreImage img{
        width: 70%;
    }
    .songsContainer{
        margin-left: -5rem;
    }
}