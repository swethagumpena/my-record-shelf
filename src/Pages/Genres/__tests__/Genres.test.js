import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Genres from '../Genres';

describe(Genres.name, () => {
  const mockGenre = {
    Bollywood:
    [
      {
        albumArtUrl: 'https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6',
        artist:
            {
              id: 'b22d7ed5-3ded-4958-a036-60453b7fc102',
              name: 'Neha Kakkar',
            },
        genre:
            {
              id: '37146752-2c95-4204-a845-a32f3cdce9c8',
              name: 'Bollywood',
            },
        id: '07075cdb-9e5c-41ec-8c09-9a1322dd92d8',
        name: 'Aankh Marey (From Simmba)',
        publishedAt: '2018-12-06T00:00:00',
      },
    ],
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot', () => {
    const { container } = render(<Genres filteredSongs={mockGenre} />);
    expect(container).toMatchSnapshot();
  });

  test('should display all songs', () => {
    render(<Genres filteredSongs={mockGenre} />);
    expect(screen.getByText('genres'));
  });

  test('should display a X button', () => {
    render(<BrowserRouter><Genres filteredSongs={mockGenre} /></BrowserRouter>);
    const songElement = screen.getByText('X');
    expect(songElement.tagName).toBe('BUTTON');
    fireEvent.click(songElement);
    expect(document.location.href).toBe('http://localhost/songs');
  });
});
