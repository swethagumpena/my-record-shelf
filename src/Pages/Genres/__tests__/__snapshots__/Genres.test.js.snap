// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres should match snapshot 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <p
        class="heading"
      >
        genres
      </p>
      <button
        class="toggle"
        type="button"
      >
        X
      </button>
    </div>
    <div
      class="songs"
      data-testid="home"
    >
      <div>
        <div
          class="genreHeader"
        >
          <div
            class="genreImage"
          >
            <img
              alt="genre-img"
              src="assets/genre-Bollywood.png"
            />
          </div>
          <div
            class="songsContainer"
          >
            Bollywood
          </div>
        </div>
        <div
          class="card"
        >
          <div
            class="card"
          >
            <img
              alt="song"
              data-testid="song-image"
              src="https://i.scdn.co/image/ab67616d0000b273dc1f496d2a8d75f7fda092b6"
            />
            <div
              class="content"
            >
              <div
                class="text"
              >
                <p
                  class="song"
                >
                  Aankh Marey (From Simmba)
                </p>
                <p
                  class="artist"
                >
                  Neha Kakkar
                </p>
              </div>
              <div
                class="likeButton"
              >
                <button
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
